frontend/src/layout/components/Sidebar/index.vue

```diff
@@ -133,7 +133,7 @@ function getCheckedLabels(json: Node): string[] {
 
 const search = async () => {
     await checkIsSystemIntl();
-    let checkedLabels: string | any[];
+    let checkedLabels: any[] = [];
     if (!globalStore.isIntl) {
         const res = await getSettingInfo();
         const json: Node = JSON.parse(res.data.xpackHideMenu);
```

Comment:
The code change seems mostly correct, but here's a quick review:

1. **Type of `checkedLabels`:** The line `let checkedLabels: string | any[];` specifies that `checkedLabels` can be either a string or an array of any type (`any[]`). Ensure that this is intentional as it might lead to runtime errors if you expect `checkedLabels` to always be a string when used.

2. **Potential Issue with `globalStore.isIntl`:** Before assigning the value from `getSettingInfo`, ensure there isn't a race condition where `globalStore.isIntl` hasn't been initialized yet (e.g., using asynchronous operations outside this block).

3. **Suggested Change:** Consider initializing `checkedLabels` within the same scope as its definition without specifying types first, especially if types are not strictly necessary for all use cases (although they can help improve readability):

   ```javascript
   const search = async () => {
       await checkIsSystemIntl();
       let checkedLabels;

       if (!globalStore.isIntl) {
           const res = await getSettingInfo();
           globalStore.setLocale(res.data.xpackHideMenu); // Assuming setLocale updates isCheckedLabels based on xpackHideMenu data

           checkedLabels = globalStore.isCheckedLabels || [];
       } else {
           checkedLabels = []; // Alternatively set labels based on intl system settings or default values
       }

       doSomethingWith(checkedLabels);
   };
   ```

This approach assumes `setCheckedLabels` method exists in `globalStore`.

4. If `isCheckedLabels` doesn’t exist, consider setting its initial value in `init` function before calling `search`. This would prevent undefined behavior:

   ```javascript
   import { init } from "./store";

   init() {
       globalStore.isCheckedLabels = []; // Set initial state if needed
   }

   async search() {
       await ...
   }
   ```

These optimizations focus on clarity and ensuring the flow is smooth without introducing unnecessary complexity or runtime error risks.